bin/warmup

```diff
@@ -0,0 +1,275 @@
+#!/usr/bin/env python2.7
+
+import argparse
+import json
+import logging
+import os
+import os.path
+import subprocess
+import sys
+
+from distutils.spawn import find_executable
+from logging import debug, error, info, warn
+sys.path.insert(0, os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))
+from warmup.krun_results import csv_to_krun_json, parse_krun_file_with_changepoints
+from warmup.summary_statistics import collect_summary_statistics, convert_to_latex, write_latex_table
+
+
+ABS_TIME_FORMAT = '%Y-%m-%d %H:%M:%S'
+BINDIR = os.path.dirname(os.path.abspath(__file__))
+DEFAULT_WINDOW_RATIO = 0.1
+DEFAULT_STEADY_RATIO = 0.25
+SCRIPT_MARK_OUTLIERS = os.path.join(BINDIR, 'mark_outliers_in_json')
+SCRIPT_MARK_CHANGEPOINTS = os.path.join(BINDIR, 'mark_changepoints_in_json')
+SCRIPT_PLOT_KRUN_RESULTS = os.path.join(BINDIR, 'plot_krun_results')
+
+CONSOLE_FORMATTER = PLAIN_FORMATTER = logging.Formatter(
+    '[%(asctime)s: %(levelname)s] %(message)s',
+    ABS_TIME_FORMAT)
+try:
+    import colorlog
+    CONSOLE_FORMATTER = colorlog.ColoredFormatter(
+        "%(log_color)s[%(asctime)s %(levelname)s] %(message)s%(reset)s",
+        ABS_TIME_FORMAT)
+except ImportError:
+    pass
+
+
+def fatal(msg):
+    """Log error and exit."""
+    error(msg)
+    sys.exit(1)
+
+
+def create_arg_parser():
+    script = os.path.basename(__file__)
+    description = ('Analyse CSV results file and produce JSON, HTML (table) or '
+                   'PDF plots. The CSV input file(s) should be in this format:\n\n'
+                   '\tprocess num, bench_name, 0, 1, 2, ...\n'
+                   '\t0, spectral norm, 0.2, 0.1, 0.4, ...\n'
+                   '\t1, spectral norm, 0.3, 0.15, 0.2, ...\n'
+                   '\n\nExample usage:\n\n\t$ python %s results.csv --output-json summary.json '
```

Comment:
It would probably be more conventional to pull all the args first, then `results.csv` at the end of the command? We can also possibly reuse the `uname` trick mentioned elsewhere.